File: app/src/main/java/com/kakacat/minitool/inquireip/Contract.java
package com.kakacat.minitool.inquireip;

import com.kakacat.minitool.common.base.IPresenter;
import com.kakacat.minitool.common.base.IView;

public interface Contract {

    interface Presenter extends IPresenter {
        void requestIpData(String input);
    }

    interface View extends IView<Presenter> {
        void onUpdateDataCallBack(IpBean ipBean, String result);
    }
}

File: app/src/main/java/com/kakacat/minitool/inquireip/Contract.kt
package com.kakacat.minitool.inquireip

import com.kakacat.minitool.common.base.IPresenter
import com.kakacat.minitool.common.base.IView

interface Contract {
    interface Presenter : IPresenter {
        fun requestIpData(input: String?)
    }

    interface View : IView<Presenter?> {
        fun onUpdateDataCallBack(ipBean: IpBean?, result: String?)
    }
}